lib/gems/pending/util/require_with_logging.rb

```diff
@@ -24,22 +26,33 @@ module Kernel
   }
   REQ_LOG_TREE = "| "
 
-  def log_require(path, mode, timing = nil)
+  def require_with_metrics
+    initial_memory = Sys::ProcTable.ps(:pid => Process.pid).rss
+    initial_loaded = $LOADED_FEATURES.length
+
+    result = yield
+
+    memory_after = Sys::ProcTable.ps(:pid => Process.pid).rss
```

Comment:
Should there be some GC calls in here to be more accurate?  Or is this relatively accurate since the calls are so close to one another?

Reply:
I considered it but I think it's fairly obvious memory snapshots are supposed to be estimates.  I can if you think it's worth it.

Reply:
Nah it's fine - just thought I'd raise the discussion.

Reply:
Note, I tried doing `GC.start` at the start of the method and it's super slow.  I think it's better to assume the memory number is a very rough estimate.